libbindgen/src/codegen/mod.rs

```diff
@@ -274,22 +269,28 @@ impl CodeGenerator for Module {
                item: &Item) {
         debug!("<Module as CodeGenerator>::codegen: item = {:?}", item);
 
-        if !ctx.options().enable_cxx_namespaces {
-            for child in self.children() {
-                ctx.resolve_item(*child)
-                    .codegen(ctx, result, whitelisted_items, &());
+        let codegen_self = |result: &mut CodegenResult| {
+            // FIXME: This could be less expensive, I guess.
```

Comment:
Nope, sorry, I remember why I did this now. Note that the `children` items only contains stuff parsed directly under that namespace, but not stuff like specializations, and base clases that we parse on `CXXBaseSpecifier`.

Makes sense to make ns.children be equivalent to every children that has the namespace as direct parent, so I rework how children are collected (I'd prefer it to be in a followup though).